### PR TITLE
Remove CUSPATIAL_BUILD_WHEELS and standardize Python builds

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Some minor simplification in advance of the scikit-build-core migration to better align wheel and non-wheel Python builds. Additionally, this PR simplifies handling of the nvcomp targets and their installation.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cuspatial/CMakeLists.txt [hunk 1]
```diff
@@ -32,52 +32,32 @@ project(
...
-  endif()
-
-  # Since there are multiple subpackages of cuspatial._lib that require access to libcuspatial, we place the
-  # library in the cuspatial directory as a single source of truth and modify the other rpaths
-  # appropriately.
-  install(TARGETS cuspatial DESTINATION ${cython_lib_dir})
+  include(cmake/Modules/WheelHelpers.cmake)
+  # TODO: This install is currently overzealous. We should only install the libraries that are
+  # downloaded by CPM during the build, not libraries that were found on the system.  However, in
+  # practice right this would only be a problem is if libcudf was not found but some of the
```
Comment (harrism) [inline python/cuspatial/CMakeLists.txt:55]:
```suggestion
  # practice this would only be a problem if libcudf was not found but some of the
```

**Review by harrism — APPROVED** [review]

**Review by raydouglass — APPROVED** [review]

File: python/cuspatial/CMakeLists.txt [hunk 2]
```diff
@@ -44,40 +42,22 @@ endif()
...
-  set(_exclude_from_all "")
-  if(CUSPATIAL_BUILD_WHEELS)
-
-    # Statically link cudart if building wheels
-    set(CUDA_STATIC_RUNTIME ON)
-    set(CUSPATIAL_USE_CUDF_STATIC ON)
-    set(CUSPATIAL_EXCLUDE_CUDF_FROM_ALL ON)
-
-    # Always build wheels against the pyarrow libarrow.
-    set(USE_LIBARROW_FROM_PYARROW ON)
```
Comment (trxcllnt) [inline python/cuspatial/CMakeLists.txt:56]: @vyasr This causes cuDF's CMake to [build and install a second libarrow](https://github.com/rapidsai/cuspatial/actions/runs/7118401836/job/19381328724#step:9:719). Don't we need to link to pyarrow's `libarrow.so` due to the internal globals?

Comment (vyasr) [reply python/cuspatial/CMakeLists.txt:56]: Yes, this was an error on my part. I went through a couple of iterations of updating the cudf CMake and at one point I had made this detection automatic within cudf. However the other unifications here are desirable, so we just want to specify that variable. Fixed in #1334.

**Pull request merged**